### PR TITLE
hud: keystroke in hud to open url in browser [ch523]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @landism, @yuindustries,

Please review the following commits I made in branch maiamcc/open-browser:

6c67bb1e63a03fa4958a655a4694d9424bcdc5fe (2018-10-18 16:09:32 -0400)
hud: keystroke in hud to open url in browser [ch523]

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/hud/renderer.go [hunk 1]
```diff
@@ -94,7 +95,7 @@ var podStatusColors = map[string]tcell.Color{
 	"CrashLoopBackOff":  cBad,
 }
 
-func (r *Renderer) layout(v view.View) rty.Component {
+func (r *Renderer) layout(v view.View) (c rty.Component, selectedResource view.Resource) {
```
Comment (maiamcc) [inline internal/hud/renderer.go]: not positive about the best way to find/store the selected resource... seems wrong to store it on the render? this is the best i've got -- does this seem right to ppl?

File: internal/hud/hud.go [hunk 2]
```diff
@@ -113,6 +115,19 @@ func (h *Hud) handleScreenEvent(ctx context.Context, st *store.Store, ev tcell.E
...
+				// TODO(nick): We might need some hints on what load balancer to
+				// open if we have multiple, or what path to default to on the opened manifest.
+				if len(h.selectedResource.Endpoints) > 0 {
+					err := browser.OpenURL(h.selectedResource.Endpoints[0])
+					if err != nil {
+						logger.Get(ctx).Infof("error opening url '%s' for resource '%s': %v",
+							h.selectedResource.Endpoints[0], h.selectedResource.Name, err)
+					}
+				} else {
+					logger.Get(ctx).Infof("no urls for resource '%s' ¯\\_(ツ)_/¯", h.selectedResource.Name)
```
Comment (maiamcc) [inline internal/hud/hud.go]: do we actually need this log line? should it be `debug` instead?

File: internal/hud/renderer.go (hunk 1 above)
Comment (dbentley) [reply internal/hud/renderer.go]: Argh, I'm sorry. Yeah, we should rethink the structure of how to store state after a render.

**Comment by maiamcc** [comment]
alright @yuindustries @landism, ready! (@dbentley better?)

**Pull request merged**